src/main/java/uk/gov/companieshouse/web/accounts/service/notehandler/dates/smallfull/TangibleAssetsDateHandler.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.companieshouse.web.accounts.service.notehandler.dates.smallfull;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.model.accounts.smallfull.SmallFullApi;
+import uk.gov.companieshouse.web.accounts.enumeration.NoteType;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.smallfull.notes.tangible.TangibleAssets;
+import uk.gov.companieshouse.web.accounts.service.notehandler.dates.DateHandler;
+import uk.gov.companieshouse.web.accounts.service.smallfull.SmallFullService;
+
+@Component
+public class TangibleAssetsDateHandler implements DateHandler<TangibleAssets> {
+
+    @Autowired
+    private SmallFullService smallFullService;
+
+    @Override
+    public void addDates(ApiClient apiClient, String transactionId, String companyAccountsId, TangibleAssets note) throws ServiceException {
+
+        SmallFullApi smallFullApi = smallFullService.getSmallFullAccounts(apiClient, transactionId, companyAccountsId);
+
+        note.setLastAccountsPeriodEndOn(smallFullApi.getLastAccounts().getPeriodEndOn());
```

Comment:
Careful on this one - if a user is filing for a new company, they won't have `lastAccounts` and this will throw an NPE. Before this, we wrapped this in an Optional#map